cli/cmd/publicendpoint.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"encoding/json"
+    "fmt"
+	"os"
+    "strconv"
+
+	"github.com/codegangsta/cli"
+	"github.com/control-center/serviced/domain/service"
+)
+
+// The vhost and port public endpoint structures are different, so we'll
+// make a unified structure for output (both text and json) that matches
+// the UI table.  This is only needed for output, not for api commands.
+type PublicEndpoint struct {
+    Service     string
+    ServiceID   string
+    Application string
+    EpType      string
+    Protocol    string
+    Name        string
+    Enabled     bool
+}
+
+func NewPublicEndpoint(service string, serviceID string, endpoint string, epType string,
+                           protocol string, name string, enabled bool) PublicEndpoint {
+	return PublicEndpoint{
+        Service     : service,
+        ServiceID   : serviceID,
+        Application : endpoint,
+        EpType      : epType,
+        Protocol    : protocol,
+        Name        : name,
+        Enabled     : enabled,
+	}
+}
+
+// serviced service public-endpoints
+func (c *ServicedCli) cmdPublicEndpointsList(ctx *cli.Context) {
```

Comment:
Naming - the _ prefix is not a convention I've seen us use for the GO code. The fact that the method name is lower case already makes it package private.  I suggest changing `cmdPublicEndpointsList` to `cmdPublicEndpointsListAll` and then change `_cmdPublicEndpointsList` to just `cmdPublicEndpointsList`